src/WorldStorage/NBTChunkSerializer.cpp

```diff
@@ -104,7 +104,8 @@ void cNBTChunkSerializer::AddItem(const cItem & a_Item, int a_Slot, const AStrin
 	// Write the tag compound (for enchantment, firework, custom name and repair cost):
 	if (
 		(!a_Item.m_Enchantments.IsEmpty()) ||
-		((a_Item.m_ItemType == E_ITEM_FIREWORK_ROCKET) || (a_Item.m_ItemType == E_ITEM_FIREWORK_STAR)) ||
+		((a_Item.m_ItemType == E_ITEM_FIREWORK_ROCKET) || (a_Item.m_ItemType == E_ITEM_FIREWORK_STAR) ||
```

Comment:
At the very least the double-opening-parenthesis is mismatched, closing in between the two conditions for books. I'd vote for removing it as well (but keep the firework stuff on one line and the book stuff on another line.